test/unit/test_provenance.py

```diff
@@ -0,0 +1,63 @@
+import os
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+
+from autosubmit.autosubmit import Autosubmit
+from log.log import AutosubmitCritical
+
+@pytest.fixture
```

Comment:
Two blank lines expected as per pep8 (two lines in a file, one blank line for functions within a `class`). The rest of this code looks excellent!